src/core/session.py

```diff
@@ -139,21 +193,76 @@ def init(self, directory: str, load: bool = True) -> None:
         # read project
         if load:
             QgsProject.instance().read(self.db.projectUri)
+            self._environmentLayer.addSoundAction(self.environmentLayer)
+            self._sightingsLayer.addSoundAction(self.sightingsLayer)
+            self._followersLayer.addSoundAction(self.followersLayer)
             QgsSettings().setValue("qgis/enableMacros", "SessionOnly")
+            self.environmentLayer.attributeValueChanged.connect(
+                self.updateRouteTypeStatus
+            )
 
     def addEnvironmentFeature(self) -> QgsVectorLayer:
         layer = self.environmentLayer
-        self._addFeature(layer)
+        self._addFeature(
+            layer,
+            geom=self.lastGpsGeom,
+            status=int(bool(layer.featureCount())),
+        )
         return layer
 
+    def updateRouteTypeStatus(self, fid, idx, value) -> None:
```

Comment:
typing

Reply:
moreover, can you add some comments tin this method please? Because I think like it's not the last time that we've to work on this...